lib/membrane_rtmp_plugin/rtmp/sink/sink.ex

```diff
@@ -0,0 +1,197 @@
+    sr_index = AAC.sample_rate_to_sampling_frequency_id(caps.sample_rate)
+    channel_configuration = AAC.channels_to_channel_config_id(caps.channels)
+    frame_length_id = AAC.samples_per_frame_to_frame_length_id(caps.samples_per_frame)
+
+    aac_config =
+      <<profile::5, sr_index::4, channel_configuration::4, frame_length_id::1, 0::1, 0::1>>
+
+    case Native.init_audio_stream(state.native, caps.channels, caps.sample_rate, aac_config) do
+      {:ok, ready, native} ->
+        Membrane.Logger.debug("Correctly initialized audio stream.")
+        state = Map.merge(state, %{native: native, ready: ready})
+        {:ok, state}
+
+      {:error, :caps_resent} ->
+        Membrane.Logger.error(
+          "Input caps redefined on pas :audio. RTMP Sink does not support changing stream paremeters"
+        )
+
+        {:ok, state}
+    end
+  end
+
+  @impl true
+  def handle_write(pad, buffer, _ctx, %{ready: false} = state) do
+    state = Map.put(state, :buffered_frame, {pad, buffer})
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_write(
+        pad,
+        buffer,
+        ctx,
+        %{ready: true, buffered_frame: {frame_pad, frame}} = state
+      ) do
+    {_, state} = handle_write(frame_pad, frame, ctx, Map.delete(state, :buffered_frame))
+    {_, state} = handle_write(pad, buffer, ctx, state)
+    {{:ok, demand: state.current_timestamps |> Enum.min_by(&elem(&1, 1)) |> elem(0)}, state}
+  end
+
+  @impl true
+  def handle_write(:video, buffer, _ctx, %{ready: true} = state) do
+    case Native.write_video_frame(
+           state.native,
+           buffer.payload,
+           buffer.dts,
+           buffer.metadata.h264.key_frame?
+         ) do
+      {:ok, native} ->
+        state = Map.put(state, :native, native)
+
+        state =
+          Map.update!(state, :current_timestamps, fn curr_tmps ->
+            Map.put(curr_tmps, :video, buffer.dts)
+          end)
+
+        {{:ok, demand: state.current_timestamps |> Enum.min_by(&elem(&1, 1)) |> elem(0)}, state}
+
+      {:error, reason} ->
+        raise("Writing video frame failed with reason: #{reason}")
+    end
+  end
+
+  @impl true
+  def handle_write(:audio, buffer, _ctx, %{ready: true} = state) do
+    buffer_pts = buffer.pts |> Ratio.ceil()
+
+    case Native.write_audio_frame(state.native, buffer.payload, buffer_pts) do
+      {:ok, native} ->
+        state = Map.put(state, :native, native)
+
+        state =
+          Map.update!(state, :current_timestamps, fn curr_tmps ->
+            Map.put(curr_tmps, :audio, buffer_pts)
+          end)
+
+        {{:ok, demand: state.current_timestamps |> Enum.min_by(&elem(&1, 1)) |> elem(0)}, state}
+
+      {:error, reason} ->
+        raise("Writing audio frame failed with reason: #{reason}")
+    end
+  end
+
+  @impl true
+  def handle_end_of_stream(
+        pad,
+        _ctx,
+        %{current_timestamps: %{audio: _audio_dts, video: _video_dts} = curr_tmps} = state
```

Comment:
Don't use abbreviations when not necessary. Maybe moving that to the function body would be more readable

Reply:
This function clause will be entirely refactored (and this abbreviation will be removed) as requested by @daniel-jodlos.